alerting/src/main/kotlin/org/opensearch/alerting/util/ClusterMetricsCoordinator.kt

```diff
@@ -0,0 +1,386 @@
+    private val clusterService: ClusterService,
+    private val threadPool: ThreadPool
+) : ClusterStateListener, CoroutineScope, LifecycleListener() {
+
+    @Volatile private var metricsExecutionFrequency = METRICS_EXECUTION_FREQUENCY.get(settings)
+    @Volatile private var metricsStoreTime = METRICS_STORE_TIME.get(settings)
+    private var dataPointCollectionDeletionJob: Scheduler.Cancellable? = null
+    companion object {
+        @Volatile
+        var isRunningFlag = false
+            internal set
+        var isDeletionUpdated = false
+            internal set
+        var isCollectionUpdated = false
+            internal set
+    }
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default + CoroutineName("ClusterMetricsCoordinator")
+
+    init {
+        clusterService.addListener(this)
+        clusterService.addLifecycleListener(this)
+        clusterService.clusterSettings.addSettingsUpdateConsumer(METRICS_EXECUTION_FREQUENCY) {
+            metricsExecutionFrequency = it
+            isCollectionUpdated = true
+        }
+        clusterService.clusterSettings.addSettingsUpdateConsumer(METRICS_STORE_TIME) {
+            metricsStoreTime = it
+            isDeletionUpdated = true
+        }
+    }
+
+    override fun clusterChanged(event: ClusterChangedEvent?) {
+        val scheduledJobCollection = Runnable {
+            launch {
+                createDocs(client as NodeClient, clusterService)
+                deleteDocs(client)
+            }
+        }
+        if (isCollectionUpdated || isDeletionUpdated) {
+            dataPointCollectionDeletionJob?.cancel()
+            log.info("Cancelled data collection and deletion jobs")
+            isRunningFlag = false
+            log.info("detected changes to settings, resetting running, deletion and collection flags to false")
+            isDeletionUpdated = false
+            isCollectionUpdated = false
+        }
+        if (event!!.localNodeMaster() && !isRunningFlag) {
+            log.info("cluster changed metricsExecutionFrequency = $metricsExecutionFrequency")
+            dataPointCollectionDeletionJob = threadPool.scheduleWithFixedDelay(
+                scheduledJobCollection,
+                metricsExecutionFrequency,
+                ThreadPool.Names.SYSTEM_WRITE
+            )
+            isRunningFlag = true
+        }
+    }
+
+    private suspend fun createDocs(client: NodeClient, clusterService: ClusterService) {
+        val currentTime = Instant.now().toString()
+        log.info("This is the current time: $currentTime")
+        val clusterHealth = client.admin().cluster().health(ClusterHealthRequest()).get().toMap()
+        val nodeStats = client.admin().cluster().nodesStats(NodesStatsRequest().addMetrics("process", "jvm")).get().toMap()
+
+        ClusterMetricsVisualizationIndex.initFunc(client, clusterService)
+
+        val unassignedShards = clusterHealth["unassigned_shards"].toString()
```

Comment:
Instead of typing out metric types like `unassigned_shards` in these lines, can you refactor them to instead call your enum class?

Reply:
I did this for all except cluster status, since the response object has "status" in it.

Reply:
CPU usage and JVM pressure as well, since the response body does not return "cpu_usage", "jvm_pressure" explicitly.